src/status_im2/contexts/chat/messages/content/reactions/view.cljs

```diff
@@ -5,31 +5,56 @@
             [utils.re-frame :as rf]
             [status-im2.contexts.chat.messages.drawers.view :as drawers]))
 
-(defn message-reactions-row
-  [chat-id message-id]
-  (let [reactions (rf/sub [:chats/message-reactions message-id chat-id])]
-    (when (seq reactions)
-      [rn/view {:margin-left 52 :margin-bottom 12 :flex-direction :row}
-       (for [{:keys [own emoji-id quantity emoji-reaction-id] :as emoji-reaction} reactions]
-         ^{:key (str emoji-reaction)}
-         [rn/view {:style {:margin-right 6}}
-          [quo/reaction
-           {:emoji               (get constants/reactions emoji-id)
-            :neutral?            own
-            :clicks              quantity
-            :on-press            (if own
-                                   #(rf/dispatch [:models.reactions/send-emoji-reaction-retraction
-                                                  {:message-id        message-id
-                                                   :emoji-id          emoji-id
-                                                   :emoji-reaction-id emoji-reaction-id}])
-                                   #(rf/dispatch [:models.reactions/send-emoji-reaction
+(defn message-reactions-row-comp
+  [{:keys [message-id chat-id]} user-message-content show-reaction-author-list?]
+  (let [reactions        (rf/sub [:chats/message-reactions message-id chat-id])
+        reaction-authors (rf/sub [:chats/reaction-authors-list chat-id])]
+    (rn/use-effect (fn []
```

Comment:
this is not good for 2 reasons, we have logic in the view, but views must be as dumb as possible all logic should be in event, and as i understand from the code we want to show changes in the bottom sheet if there is new reaction, but there is an animation, and it will be bad UX, we should make drawer reactive
and its better to avoid using effect and functional components if its possible

Reply:
We don't know to make the emoji reaction count reactive because the data is fetched only on `long-press` of reaction, So it won't be updated with time.

Reply:
What happens here is I am trying to only show the sheet once if the user long pressed a reaction, But we don't know how long the data will take to get fetched from status-go, So that's why I added the effect.

Reply:
If we wanted to show the sheet after the data is received from status-go in the events namespace, we'd have to require the `drawers` namespace and the `user-message` UI component (To be used as the `selected-item` of bottom sheet).

Is there a way around having logic in view or view in logic?
CC: @ilmotta

Reply:
@ibrkhalil, isn't the emoji reaction count readily available in status-go? What I mean is that from status-go point of view there's no network roundtrip to get emoji reactions (it's all available in the SQLite DB), so it should be nearly instant to get the response from an RPC request in status-mobile.

If what I said above is true, then I think it should be perfectly fine to move the logic to the event layer, in terms of performance and UX. That is, when the user long-presses on a reaction, a simple event is dispatched. This event in turn makes an RPC request to get reaction counters. The success result of the RPC request then changes the app db, and, finally, the subscription on the view is changed and the view re-renders. The usual re-frame dance.

So that should be aligned with what @flexsurfer's suggested, and it should give a good UX too.

Reply:
> If we wanted to show the sheet after the data is received from status-go in the events namespace, we'd have to require the drawers namespace and the user-message UI component (To be used as the selected-item of bottom sheet).

Great question @ibrkhalil, I think this is somewhat unavoidable in all frontend apps I worked with, in some level... But, having said that, usually if this situation arises we should think back on the app state, because there's usually a way to decouple the UI and event layer. For example, maybe the `selected-item` should not be local state and should be handled by re-frame, etc.

Reply:
Done, I added a conditional to check if the data exists.